microservices/uyuni-java-common/src/main/java/com/suse/common/configuration/FileConfigurationSource.java

```diff
@@ -0,0 +1,213 @@
+    // then we need to do a lexicographical comparison to make sure that the filenames themselves are different.
+    private static final Comparator<Path> CONFIG_PATH_COMPARATOR = (p1, p2) -> {
+        int lenDif = p2.toAbsolutePath().toString().length() - p1.toAbsolutePath().toString().length();
+        return lenDif != 0 ? lenDif : p2.compareTo(p1);
+    };
+
+    private static final Logger LOGGER = LogManager.getLogger(FileConfigurationSource.class);
+
+    private final List<String> fallbackNamespaces;
+
+    private final String commonFilePrefix;
+
+    private final Properties configValues;
+
+    /**
+     * Builds a file configuration source.
+     * @param fileSourcesIn List of files and directories to process
+     */
+    public FileConfigurationSource(List<Path> fileSourcesIn) {
+        this(fileSourcesIn, null, Collections.emptyList());
+    }
+
+    /**
+     * Builds a file configuration source.
+     * @param fileSourcesIn List of files and directories to process
+     * @param filePrefixIn a common prefix used by all configuration file names
+     */
+    public FileConfigurationSource(List<Path> fileSourcesIn, String filePrefixIn) {
+        this(fileSourcesIn, filePrefixIn, Collections.emptyList());
+    }
+
+    /**
+     * Builds a file configuration source.
+     * @param fileSourcesIn List of files and directories to process
+     * @param filePrefixIn a common prefix used by all configuration file names
+     * @param fallbackNamespacesIn namespaces to search for, in the given order. These are used when a property,
+     *     specified with no namespace, was not found.
+     */
+    public FileConfigurationSource(List<Path> fileSourcesIn, String filePrefixIn, List<String> fallbackNamespacesIn) {
+        configValues = new Properties();
+
+        commonFilePrefix = filePrefixIn;
+        fallbackNamespaces = fallbackNamespacesIn;
+
+        fileSourcesIn.stream()
+            .flatMap(path -> getConfigurationFiles(path))
+            .sorted(CONFIG_PATH_COMPARATOR)
+            .map(file -> loadProperties(file))
+            .forEach(props -> configValues.putAll(props));
+    }
+
+    /**
+     * Get the configuration entry for given property
+     *
+     * @param property string to get the value of
+     * @return the value
+     */
+    @Override
+    protected String getRawValue(String property) {
+        LOGGER.debug("getString() - called with: {}", property);
+        if (property == null) {
+            return null;
+        }
+
+        String namespace = "";
+        String propertyName = property;
+
+        int lastDot = property.lastIndexOf('.');
+        if (lastDot > 0) {
+            propertyName = property.substring(lastDot + 1);
+            namespace = property.substring(0, lastDot);
+        }
+
+        LOGGER.debug("getString() - Getting property: {}", propertyName);
+        String result = configValues.getProperty(propertyName);
```

Comment:
If a property is defined without a namespace, it will always override the namespaced one. That's the current behaviour of our configuration logic. Check the [test `testOverride()` in original `ConfigTest`](https://github.com/uyuni-project/uyuni/blob/f8ee365162741cc8172794de08b8ed9500b4fc43/java/code/src/com/redhat/rhn/common/conf/test/ConfigTest.java#L102-L110).